src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Components.Wasm.targets

```diff
@@ -151,6 +152,30 @@ Copyright (c) .NET Foundation. All rights reserved.
   </Target>
 
   <Target Name="_ProcessBlazorWasmOutputs" DependsOnTargets="_ResolveBlazorWasmOutputs">
+    <PropertyGroup>
+      <_BlazorBuildGZipCompressDirectory>$(IntermediateOutputPath)build-gz\</_BlazorBuildGZipCompressDirectory>
+    </PropertyGroup>
+
+    <!--
+      Compress referenced binaries using GZip during build. This skips files such as the project's assemblies
+      that change from build to build. Runtime assets contribute to the bulk of the download size. Compressing it
+      has the most benefit while avoiding any ongoing costs to the dev inner loop.
```

Comment:
Does "skip the project's own assemblies" really make the build perceptibly faster? I would have thought that gzipping a few hundred KB (for a fairly large app assembly) would be basically instant.

Not suggesting blocking this PR on that question. I just want to understand.

Reply:
I thought I had numbers that showed this was going be expensive, but calling GZip on most .NET assemblies, including ones that are several 100kb, seems to be in the order of a few ms. There's a few outliers (like S.T.Encodings takes about 20-30ms for a < 1MB input file), but a few 100kb sized blazor app is a blip.

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/23655 so we could consider this.